providers/google/tests/unit/google/cloud/hooks/test_cloud_sql.py

```diff
@@ -761,13 +768,40 @@ def test_delete_database_overridden_project_id(
         )
 
 
+def _parse_from_uri(uri: str):
+    connection_parameters = {}
+    uri_parts = urlsplit(uri)
+    connection_parameters["conn_type"] = uri_parts.scheme
+    rest_of_the_url = uri.replace(f"{uri_parts.scheme}://", "//")
+    uri_parts = urlsplit(rest_of_the_url)
+    host = unquote(uri_parts.hostname or "")
+    connection_parameters["host"] = host
+    quoted_schema = uri_parts.path[1:]
+    connection_parameters["schema"] = unquote(quoted_schema) if quoted_schema else ""
+    connection_parameters["login"] = unquote(uri_parts.username) if uri_parts.username else ""
+    connection_parameters["password"] = unquote(uri_parts.password) if uri_parts.password else ""
+    connection_parameters["port"] = uri_parts.port  # type: ignore[assignment]
+    if uri_parts.query:
+        query = dict(parse_qsl(uri_parts.query, keep_blank_values=True))
+        connection_parameters["extra"] = json.dumps(query)
+    return connection_parameters
+
+
 class TestCloudSqlDatabaseHook:
     @mock.patch("airflow.providers.google.cloud.hooks.cloud_sql.CloudSQLDatabaseHook.get_connection")
     def test_cloudsql_database_hook_validate_ssl_certs_no_ssl(self, get_connection):
-        connection = Connection()
-        connection.set_extra(
-            json.dumps({"location": "test", "instance": "instance", "database_type": "postgres"})
+        connection = Connection(
+            conn_id="my_test_connection",
+            conn_type="google_cloud_platform",
```

Comment:
@Crowiant I am not sure that `google_cloud_platform ` is a correct `conn_type` here I think it should be `gcpcloudsqldb`, correct me if I am wrong?

Reply:
Done